integration/test/Test/Spar.hs

```diff
@@ -26,3 +34,266 @@ testSparUserCreationInvitationTimeout = do
+    checkSparGetUserAndFindByExtId OwnDomain tok newExtId userId $ \u -> do
+      u %. "externalId" `shouldMatch` newExtId
+      (u %. "emails" >>= asList >>= assertOne >>= (%. "value")) `shouldMatch` email
+    bindResponse (getUsersId OwnDomain [userId]) $ \res -> do
+      res.status `shouldMatchInt` 200
+      u <- res.json >>= asList >>= assertOne
+      u %. "email" `shouldMatch` email
+      subject <- u %. "sso_id.subject" >>= asString
+      subject `shouldContainString` newExtId
+    bindResponse (findUsersByExternalId OwnDomain tok extId) $ \res -> do
+      res.json %. "totalResults" `shouldMatchInt` 0
+      res.json %. "Resources" `shouldMatch` ([] :: [Value])
+    pure updatedScimUser
+
+  -- Verify that updating the user's email works
+  do
+    let oldEmail = email
+    newEmail <- randomEmail
+    updatedScimUser <- setField "emails" (Array (Vector.fromList [object ["value" .= newEmail]])) scimUserWith2Updates
+    currentExtId <- updatedScimUser %. "externalId" >>= asString
+    bindResponse (updateScimUser OwnDomain tok userId updatedScimUser) $ \res -> do
+      res.status `shouldMatchInt` 200
+
+    -- before activation the old email should still be present
+    checkSparGetUserAndFindByExtId OwnDomain tok currentExtId userId $ \u -> do
+      u %. "externalId" `shouldMatch` currentExtId
+      (u %. "emails" >>= asList >>= assertOne >>= (%. "value")) `shouldMatch` newEmail
+    bindResponse (getUsersId OwnDomain [userId]) $ \res -> do
+      res.status `shouldMatchInt` 200
+      u <- res.json >>= asList >>= assertOne
+      u %. "email" `shouldMatch` oldEmail
+      subject <- u %. "sso_id.subject" >>= asString
+      subject `shouldContainString` currentExtId
+
+    -- after activation the new email should be present
+    activateEmail OwnDomain newEmail
+    checkSparGetUserAndFindByExtId OwnDomain tok currentExtId userId $ \u -> do
+      u %. "externalId" `shouldMatch` currentExtId
+      (u %. "emails" >>= asList >>= assertOne >>= (%. "value")) `shouldMatch` newEmail
+    bindResponse (getUsersId OwnDomain [userId]) $ \res -> do
+      res.status `shouldMatchInt` 200
+      u <- res.json >>= asList >>= assertOne
+      u %. "email" `shouldMatch` newEmail
+      subject <- u %. "sso_id.subject" >>= asString
+      subject `shouldContainString` currentExtId
+
+registerTestIdPWithMeta :: (HasCallStack, MakesValue owner) => owner -> App Response
+registerTestIdPWithMeta owner = do
+  SampleIdP idpmeta _ _ _ <- makeSampleIdPMetadata
+  createIdp owner idpmeta
+
+testSparExternalIdDifferentFromEmail :: (HasCallStack) => App ()
+testSparExternalIdDifferentFromEmail = do
+  (owner, tid, _) <- createTeam OwnDomain 1
+  tok <- createScimToken owner >>= \resp -> resp.json %. "token" >>= asString
+  email <- randomEmail
+  extId <- randomExternalId
+  scimUser <- randomScimUserWith extId email
+  userId <- createScimUser OwnDomain tok scimUser >>= getJSON 201 >>= (%. "id") >>= asString
+  registerUser OwnDomain tid email
```

Comment:
inspect situation *before* registration? maybe just mv L143-151 into a helper function and all that twice?